File: cs/Problems/Day8Test.cs
namespace AOC2023.Problems;

public class Day8Tests
{
    private readonly Day8 sut = new();

    [Theory, InlineData(2, 1), InlineData(6, 2)]
    public void TestSet_ShouldYield_Result(int expected, int testCase)
    {
        var input = InputReader.ReadProblemInput("day8_" + testCase);
        var result = sut.Solve(input);

        Assert.Equal(expected, result);
    }

    [Theory, InlineData(14893)]
    public void FullSet_ShouldYield_Result(int expected)
    {
        var input = InputReader.ReadProblemInput("day8_3");
        var result = sut.Solve(input);

        Assert.Equal(expected, result);
    }
}

File: ts/problems/day8.test.ts
import test, { describe } from "node:test";
import assert from "node:assert";

import { solve } from "./day8.js";
import { readProblemInput } from "../inputReader.js";

describe("Day 8", () => {
  test("test set 1 should yield expected result", () => {
    const input = readProblemInput("day8_1");
    const result = solve(input);
    assert.equal(result, 2);
  });

  test("test set 2 should yield expected result", () => {
    const input = readProblemInput("day8_2");
    const result = solve(input);
    assert.equal(result, 6);
  });

  test("full set should yield expected result", () => {
    const input = readProblemInput("day8_3");
    const result = solve(input);
    assert.equal(result, 14893);
  });
});
